parser/src/semantic/09_validate_annotations.ts

```diff
@@ -116,6 +116,12 @@ export class ValidateAnnotationsVisitor extends Visitor {
             if (!allVariables.includes(arg.name) && annotation.bodyVariable !== arg.name) {
               throw new SemanticError(`Argument '${arg.name}' is missing from the rest annotation at ${annotation.location}`);
             }
+
+            if (annotation.method === "GET" && arg.secret) {
+              throw new SemanticError(
+                `Argument marked as secret cannot be used in the path or query parts of a GET endpoint at ${annotation.location}`,
+              );
+            }
```

Comment:
Está incorreto. Está dando erro para qualquer argumento `!secret`, enquanto que só deveria dar erro se esse argumento for usado em pathVariables ou queryVariables. Checar com array.includes.